core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/util/JsonUtil.java

```diff
@@ -39,20 +38,23 @@ public class JsonUtil {
 
     private JsonUtil() {}
 
-    public static JSONArray objectsToJsonArray(Collection<?> objects) {
-        JSONArray jsonArray = new JSONArray();
+    public static JsonArray objectsToJsonArray(Collection<?> objects) {
+        JsonArrayBuilder jsonArrayBuilder = Json.createArrayBuilder();
         objects.forEach(object -> {
             try {
                 String json = OBJECT_MAPPER.writeValueAsString(object);
-                jsonArray.put(new JSONObject(json));
-            } catch (JsonProcessingException | JSONException e) {
+                try (JsonReader jsonReader = Json.createReader(new StringReader(json))) {
```

Comment:
Before you replace: 
Please mind that a *very strange* thing is proposed here in this code. We have `object`. We have `OBJECT_MAPPER` that converts this object to a JSON tree and then writes it down as a string (both within `OBJECT_MAPPER.writeValueAsString(object)`).
Then we *read it back* into a JSON tree from a string in `JsonObject jsonObject = jsonReader.readObject()`.
And then we create a new JSON array, and insert the tree into that array.
There is quite a lot of -- to put it straight -- meaningless actions going on

Reply:
done